aggregator/src/bin/collection_job_driver.rs

```diff
@@ -57,10 +58,7 @@ async fn main() -> anyhow::Result<()> {
                 ctx.config.job_driver_config.maximum_attempts_before_failure,
             ),
         ));
-        select! {
-            _ = job_driver.run() => {}
-            _ = shutdown_signal => {}
-        };
+        stopper.stop_future(job_driver.run()).await;
```

Comment:
I think it's okay for now, since it matches existing behavior, but it would be nice to wait for outstanding jobs, so we don't burn retries unnecessarily. I'll file an issue to track that.